astroquery/gemini/core.py

```diff
@@ -0,0 +1,480 @@
+"""
+
+from datetime import date
+
+import warnings
+
+import astropy
+from astropy import units
+from astropy.table import Table, MaskedColumn
+from astropy.utils.exceptions import AstropyUserWarning
+from ..utils import commons
+import numpy as np
+
+import json
+
+from ..query import BaseQuery
+from ..utils.class_or_instance import class_or_instance
+from ..utils import async_to_sync
+from . import conf
+
+__all__ = ['Observations', 'ObservationsClass']  # specifies what to import
+
+
+__valid_instruments__ = [
+    'GMOS',
+    'GMOS-N',
+    'GMOS-S',
+    'GNIRS',
+    'GRACES',
+    'NIRI',
+    'NIFS',
+    'GSAOI',
+    'F2',
+    'GPI',
+    'NICI',
+    'MICHELLE',
+    'TRECS',
+    'BHROS',
+    'HRWFS',
+    'OSCIR',
+    'FLAMINGOS',
+    'HOKUPAA+QUIRC',
+    'PHOENIX',
+    'TEXES',
+    'ABU',
+    'CIRPASS'
+]
+
+
+__valid_observation_class__ = [
+    'science',
+    'acq',
+    'progCal',
+    'dayCal',
+    'partnerCal',
+    'acqCal',
+]
+
+__valid_observation_types__ = [
+    'OBJECT',
+    'BIAS',
+    'DARK',
+    'FLAT',
+    'ARC',
+    'PINHOLE',
+    'RONCHI',
+    'CAL',
+    'FRINGE',
+    'MASK'
+]
+
+__valid_modes__ = [
+    'imaging',
+    'spectroscopy',
+    'LS',
+    'MOS',
+    'IFS'
+]
+
+__valid_adaptive_optics__ = [
+    'NOTAO',
+    'AO',
+    'NGS',
+    'LGS'
+]
+
+__valid_raw_reduced__ = [
+    'RAW',
+    'PREPARED',
+    'PROCESSED_BIAS',
+    'PROCESSED_FLAT',
+    'PROCESSED_FRINGE',
+    'PROCESSED_ARC'
+]
+
+
+class ObservationsClass(BaseQuery):
+
+    server = conf.server
+
+    def __init__(self, *args):
+        """ set some parameters """
+        # do login here
+        super().__init__()
+
+    @class_or_instance
+    def query_region(self, coordinates, radius):
+        """
+        search for Gemini observations by target on the sky.
+
+        Given a sky position and radius, returns a list of Gemini observations.
+
+        Parameters
+        ----------
+        coordinates : str or `~astropy.coordinates` object
+            The target around which to search. It may be specified as a
+            string or as the appropriate `~astropy.coordinates` object.
+        radius : str or `~astropy.units.Quantity` object, optional
+            Default 0.3 degrees.
+            The string must be parsable by `~astropy.coordinates.Angle`. The
+            appropriate `~astropy.units.Quantity` object from
+            `~astropy.units` may also be used. Defaults to 0.3 deg.
+
+        Returns
+        -------
+        response : `~astropy.table.Table`
+        """
+        return self.query_criteria(coordinates=coordinates, radius=radius)
+
+    @class_or_instance
+    def query_object(self, objectname):
+        """
+        search for Gemini observations by target on the sky.
+
+        Given a sky position and radius, returns a list of Gemini observations.
+
+        Parameters
+        ----------
+        coordinates : str or `~astropy.coordinates` object
```

Comment:
this docstring is copied from above but is no longer true.  coordinates is not an accepted parameter, nor is radius

Reply:
oops - thanks!